.github/workflows/notify-slack.yml

```diff
@@ -0,0 +1,77 @@
+---
+
+name: 'Notify slack'
+
+on:
+  workflow_call:
+    inputs:
+      application:
+        description: 'The friendly name for the application'
+        type: string
+        required: true
+
+      environment:
+        description: |
+          The environment variables to pass to the render.  Each environment
+          variable should be in format VAR=VAL with one on each line.
+          Example:
+            environment: |
+              FOO=bar
+              BAR=baz
+        type: string
+        required: false
+        default: ''
+
+      undefined:
+        description: 'If true, undefined variables will be used in templates (no error will be raised)'
+        type: string
+        required: false
+        default: 'false'
+
+      template:
+        description: 'The path to the template file to render'
+        type: string
+        required: true
+
+      version:
+        description: 'The version that is being deployed'
+        type: string
+        default: ${{ github.ref }}
+
+      type:
+        description: 'The type of action that is being performed such as build or deployment'
+        type: string
+        required: true
+
+      status:
+        description: 'The status of the deployment.  Options are started, success, and failure'
+        type: string
+        required: true
+
+    secrets:
+      slack_webhook_url:
+        description: 'The slack webhook url to post the message to'
+        required: true
+
+defaults:
+  run:
+    shell: bash
+
+jobs:
+  notify:
+    name: 'Notify'
+    runs-on: ubuntu-latest
+    environment: ${{ inputs.environment }}
+    steps:
+      - name: 'Checkout'
+        uses: actions/checkout@v3
+
+      - name: 'Test publish slack notification workflow'
+        uses: shopsmart/github-actions/.github/actions/test-notify-slack@feature/slack-notify
+        #./.github/actions/test-notify-slack
```

Comment:
We should not be calling the test-notify-slack action from our notify-slack workflow.

The notify-slack workflow is what we are calling from other application repos.
The test-notify-slack workflow is going to call the notify-slack workflow and then the test-notify-slack action.
The test-notify-slack action holds assertions to be ran after the notify-slack workflow is run.

Please take a look at another reusable workflow and their tests to see how these are orchestrated.

https://github.com/shopsmart/github-actions/blob/main/.github/workflows/test-build-ecr-image.yml